tests/run-staging/i6992/Macro_1.scala

```diff
@@ -25,7 +25,7 @@ package scala {
           case '{$x: Foo} => Expr(run(x).x)
         }
       } catch {
-        case ex: scala.internal.quoted.ScopeException =>
+        case ex: Exception if ex.getMessage == "Cannot call `scala.quoted.staging.run(...)` within a macro or another `run(...)`" =>
```

Comment:
The test is a little brittle. I guess it is temporary given the other refactoring is in process.

Reply:
Maybe I manage to create the exception class in the compiler code. Then the tests would need to have the compiler in the classpath to know that exception type. For this I must first move the implementation of `scala.internal.quoted.Expr` into the compiler which is in process.